dhis-2/dhis-api/src/main/java/org/hisp/dhis/programstageworkinglistdefinition/ProgramStageWorkingListDefinition.java

```diff
@@ -0,0 +1,142 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.programstageworkinglistdefinition;
+
+import java.util.Objects;
+
+import lombok.Setter;
+
+import org.hisp.dhis.common.BaseIdentifiableObject;
+import org.hisp.dhis.common.DxfNamespaces;
+import org.hisp.dhis.common.MetadataObject;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.translation.Translatable;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+@Setter
+@JacksonXmlRootElement( localName = "programStageWorkingListDefinition", namespace = DxfNamespaces.DXF_2_0 )
+public class ProgramStageWorkingListDefinition extends BaseIdentifiableObject
+    implements MetadataObject
+{
+
+    /**
+     * Property indicating the program of the working list
+     */
+    private Program program;
+
+    /**
+     * Property indicating the program stage of the working list
+     */
+    private ProgramStage programStage;
+
+    /**
+     * Property indicating the description of the working list
+     */
+    private String description;
+
+    /**
+     * Criteria object representing selected projections, filtering and sorting
+     * criteria in program stages
+     */
+    private ProgramStageQueryCriteria programStageQueryCriteria = new ProgramStageQueryCriteria();
+
+    @JsonProperty
+    @JsonSerialize( as = BaseIdentifiableObject.class )
+    @JacksonXmlProperty( namespace = DxfNamespaces.DXF_2_0 )
+    public Program getProgram()
+    {
+        return program;
+    }
+
+    @JsonProperty
+    @JsonSerialize( as = BaseIdentifiableObject.class )
+    @JacksonXmlProperty( namespace = DxfNamespaces.DXF_2_0 )
+    public ProgramStage getProgramStage()
+    {
+        return programStage;
+    }
+
+    @JsonProperty
+    @JacksonXmlProperty( namespace = DxfNamespaces.DXF_2_0 )
+    public String getDescription()
+    {
+        return description;
+    }
+
+    @JsonProperty
+    @JacksonXmlProperty( namespace = DxfNamespaces.DXF_2_0 )
+    @Translatable( propertyName = "description", key = "DESCRIPTION" )
+    public String getDisplayDescription()
+    {
+        return getTranslation( "DESCRIPTION", getDescription() );
+    }
+
+    @JsonProperty
+    @JacksonXmlProperty( namespace = DxfNamespaces.DXF_2_0 )
+    public ProgramStageQueryCriteria getProgramStageQueryCriteria()
+    {
+        return programStageQueryCriteria;
+    }
+
+    @Override
+    public boolean equals( Object o )
```

Comment:
Any special reason to include an override for equals and hashcode?

Reply:
If I don't include them Sonar complaints with S2160: Subclasses that add fields should override "equals"